build-system/babel-plugins/babel-plugin-transform-function-declarations/index.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const BAIL_OUT_CONDITIONS = {
+  // If this isn't a FunctionDeclaration, bail out on modification.
+  isNotFunction: (t, path) => !t.isFunctionDeclaration(path.node),
+
+  // If this FunctionDeclaration is a Generator, bail out on modification.
+  isGenerator: (t, path) => path.node.generator,
+
+  // If this FunctionDeclaration doesn't have a single return statement, bail out on modification
+  isNotSingleReturnStatment: (t, path) =>
+    !t.isBlockStatement(path.get('body')) ||
+    !t.isReturnStatement(path.get('body.body.0')),
+
+  // Since we don't know if exported members are 'new'd, bail out on modification.
+  isExported: (t, path) =>
+    path.find(
```

Comment:
```suggestion
    path.findParent(
```

Reply:
Resolved, makes sense not to check the current path node since it's already not what we're looking for.